source/core/ib_stencils.cc

```diff
@@ -0,0 +1,235 @@
+//
+// Created by lucka on 2021-05-21.
+//
+
+#include <core/ib_stencils.h>
+
+
+
+template <int dim>
+unsigned int
+IBStencils<dim>::nb_points(unsigned int order)
+{
+  // The number of points used in the stencil excluding the DOF is equal to the
+  // order.
+  unsigned int nb_points = order;
+  // In the case where the cell is used directly a=to to find the solution at
+  // the IB only one point is needed.
+  if (order > 4)
+    nb_points = 1;
+  return nb_points;
+}
+
+template <int dim>
+std::vector<double>
+IBStencils<dim>::coefficients(unsigned int order)
+{
+  // Initialize the coefficient vector
+  std::vector<double> coef;
+
+  // Fill the coefficient vector based on the order.
+  if (order == 1)
+    {
+      coef.resize(2);
+      coef[0] = 9;
+      coef[1] = -8;
+    }
+  if (order == 2)
+    {
+      coef.resize(3);
+      coef[0] = 153;
+      ;
+      coef[1] = -288;
+      coef[2] = 136;
+    }
+  if (order == 3)
+    {
+      coef.resize(4);
+      coef[0] = 2925;
+      coef[1] = -8424;
+      coef[2] = 8100;
+      coef[3] = -2600;
+    }
+  if (order == 4)
+    {
+      coef.resize(5);
+      coef[0] = 58905;
+      coef[1] = -228480;
+      coef[2] = 332640;
+      coef[3] = -215424;
+      coef[4] = 52360;
+    }
+  if (order > 4)
+    {
+      // In this case the cell is directly used to find the solution at the IB
+      // position. In this case only one point is needed (the position of the
+      // point on the IB) and its coefficient is 1.
+      coef.resize(1);
+      coef[0] = 1;
+    }
+
+  return coef;
+}
+
+template <int dim>
+std::tuple<Point<dim>, std::vector<Point<dim>>>
+IBStencils<dim>::points(unsigned int    order,
```

Comment:
IBStencils is plural or should it be singular?

Reply:
I put plural because there are many possible stencils. But in fact only one is used so  I don’t really know.

Reply:
Singular then my friend. A quick refactor